File: leetcode/java/1688. Count of Matches in Tournament.java
class Solution {
    public int numberOfMatches(int n) {
        int teams = n;
        
        if(teams == 1) {
            return 0;
        }
        
        int total = 0;
        while(teams != 1) {
            total += (teams % 2 == 0) 
                ? (teams / 2)
                : (teams - 1) / 2;
            
            teams = (teams % 2 == 0) 
                ? (teams / 2)
                : (teams - 1) / 2 + 1;
        }
        
        return total;
    }
}

File: leetcode/kotlin/1688. Count of Matches in Tournament.kt
class Solution {
    fun numberOfMatches(n: Int): Int {
        
        if (n == 1) {
            return 0
        }
        
        var teams = n
        var totalMatches = 0
        
        while (teams != 1) {
            totalMatches += if (teams % 2 == 0) teams / 2 else (teams - 1) / 2
            teams = if (teams % 2 == 0) teams / 2 else ((teams - 1) / 2 + 1)
        }
        
        return totalMatches
    }
}
